CMakeLists.txt

```diff
@@ -28,5 +31,8 @@ file(GLOB SOURCE_FILES "src/*.cpp" "src/*.h")
 # Declare target executable
 add_executable(${TARGET} ${SOURCE_FILES})
 
+# Link libraries
+target_link_libraries(${TARGET} template-library)
```

Comment:
In target_link_libraries you should use the library name from the TEMPLATE-LIBRARY variables like <XPREFIX>_LIBRARIES. Do you think we would need to pass the <XPREFIX>_CFLAGS and stuff like this from the lib to our compiler?